google-cloud-speech/samples/speech_samples.rb

```diff
@@ -0,0 +1,481 @@
+      end_time   = word.end_time.seconds + word.end_time.nanos / 1_000_000_000.0
+
+      puts "Word: #{word.word} #{start_time} #{end_time}"
+    end
+  end
+  # [END speech_sync_recognize_words]
+end
+
+def speech_sync_recognize_gcs storage_path: nil
+  # [START speech_transcribe_sync_gcs]
+  # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
+
+  require "google/cloud/speech"
+
+  speech = Google::Cloud::Speech.speech
+
+  # [START speech_ruby_migration_config_gcs]
+  config = { encoding:          :LINEAR16,
+             sample_rate_hertz: 16_000,
+             language_code:     "en-US" }
+  audio  = { uri: storage_path }
+
+  response = speech.recognize config: config, audio: audio
+  # [END speech_ruby_migration_config_gcs]
+
+  results = response.results
+
+  alternatives = results.first.alternatives
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+  end
+  # [END speech_transcribe_sync_gcs]
+end
+
+def speech_async_recognize audio_file_path: nil
+  # [START speech_transcribe_async]
+  # audio_file_path = "Path to file on which to perform speech recognition"
+
+  require "google/cloud/speech"
+
+  speech = Google::Cloud::Speech.speech
+
+  # [START speech_ruby_migration_async_response]
+  # [START speech_ruby_migration_async_request]
+  audio_file = File.binread audio_file_path
+  config     = { encoding:          :LINEAR16,
+                 sample_rate_hertz: 16_000,
+                 language_code:     "en-US" }
+  audio      = { content: audio_file }
+
+  operation = speech.long_running_recognize config: config, audio: audio
+
+  puts "Operation started"
+
+  operation.wait_until_done!
+
+  raise operation.results.message if operation.error?
+
+  results = operation.response.results
+  # [END speech_ruby_migration_async_request]
+
+  alternatives = results.first.alternatives
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+  end
+  # [END speech_ruby_migration_async_response]
+  # [END speech_transcribe_async]
+end
+
+def speech_async_recognize_gcs storage_path: nil
+  # [START speech_transcribe_async_gcs]
+  # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
+
+  require "google/cloud/speech"
+
+  speech = Google::Cloud::Speech.speech
+
+  config = { encoding:          :LINEAR16,
+             sample_rate_hertz: 16_000,
+             language_code:     "en-US" }
+  audio = { uri: storage_path }
+
+  operation = speech.long_running_recognize config: config, audio: audio
+
+  puts "Operation started"
+
+  operation.wait_until_done!
+
+  raise operation.results.message if operation.error?
+
+  results = operation.response.results
+
+  alternatives = results.first.alternatives
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+  end
+  # [END speech_transcribe_async_gcs]
+end
+
+# rubocop:disable Metrics/AbcSize
```

Comment:
@dazuma I added this to ignore rubocop, as it seemed a bit aggressive. If you don't agree I can work on improving this.

Reply:
Added another ignore to the next sample as well.

Reply:
I think that's fine.